conjure-java-jaxrs-client/src/test/java/com/palantir/conjure/java/client/jaxrs/TracerTest.java

```diff
@@ -60,26 +54,31 @@ public void before() {
     @Test
     public void testClientIsInstrumentedWithTracer() throws InterruptedException {
         server.enqueue(new MockResponse().setBody("\"server\""));
-        OpenSpan parentTrace = Tracer.startSpan("");
-        List<Map.Entry<SpanType, String>> observedSpans = Lists.newArrayList();
-        Tracer.subscribe(TracerTest.class.getName(),
-                span -> observedSpans.add(Maps.immutableEntry(span.type(), span.getOperation())));
 
+        String parentSpanId = Tracer.startSpan("").getSpanId();
         String traceId = Tracer.getTraceId();
+
+        List<String> spanIds = Lists.newArrayList();
+        List<Span> spans = Lists.newArrayList();
+        Tracer.subscribe(TracerTest.class.getName(), span -> {
+            spanIds.add(span.getSpanId());
+            spans.add(span);
+        });
+
         service.param("somevalue");
 
         Tracer.unsubscribe(TracerTest.class.getName());
-        assertThat(observedSpans, contains(
-                Maps.immutableEntry(SpanType.LOCAL, "OkHttp: acquire-limiter-enqueue"),
-                Maps.immutableEntry(SpanType.LOCAL, "OkHttp: acquire-limiter-run"),
-                Maps.immutableEntry(SpanType.LOCAL, "OkHttp: execute-enqueue"),
-                Maps.immutableEntry(SpanType.CLIENT_OUTGOING, "OkHttp: GET /{param}"),
-                Maps.immutableEntry(SpanType.LOCAL, "OkHttp: execute-run"),
-                Maps.immutableEntry(SpanType.LOCAL, "OkHttp: dispatcher")));
+
+        assertThat(spans).containsExactly(
+                span(spans.get(0), traceId, spanIds.get(2), SpanType.LOCAL, "OkHttp: limiter-enqueue"),
+                span(spans.get(1), traceId, spanIds.get(2), SpanType.LOCAL, "OkHttp: limiter-run"),
+                span(spans.get(2), traceId, parentSpanId, SpanType.LOCAL, "OkHttp: enqueue"),
+                span(spans.get(3), traceId, spanIds.get(4), SpanType.CLIENT_OUTGOING, "OkHttp: GET /{param}"),
+                span(spans.get(4), traceId, spanIds.get(1), SpanType.LOCAL, "OkHttp: dispatcher"));
```

Comment:
It seems weird to me that this span is parented by the useless `OkHttp: limiter-run`. In my opinion, `OkHttp: limiter-run` shouldn't even exist. If it does in the meantime, it should be the parent span of anything.
I also like the approach on #1069 of just removing the Dispatcher span altogether and going straight to the `CLIENT_OUTGOING` span, which should be a sibling of the `OkHttp: limiter-*` spans

Reply:
> In my opinion, `OkHttp: limiter-run` shouldn't even exist

I agree. Ideally there would just be a single span to track the time spent waiting for the limiter. But doing so requires a change to `AsyncTracer. See my comment at https://github.com/palantir/conjure-java-runtime/pull/1073#discussion_r277124326.

> I also like the approach on #1069 of just removing the Dispatcher span altogether

I don't disagree, but there needs to be some mechanism to carry the trace over to the dispatcher thread. This should probably happen before anything runs (ie not in an interceptor). I don't think it unreasonable to keep the dispatcher span since
1. We instrument all other executors in this same way
2. There is value in tracking the duration of tasks running in this executor. This helps us know if we are spending a lot of time in a dispatcher thread doing work outside the request span.

Reply:
My first point is that the second best thing for `OkHttp: limiter-run` not existing is `OkHttp: limiter-run` being a no-op and having no children. We can always remove it later once we fix `AsyncTrancer`.